### ISSUE TITLE
Truncation error

### ISSUE THREAD

**Issue by nitayalon** [scipy/scipy#12313]
I'm using scipy.stats.truncnorm and updating the lower bound iteratively. The problem is that once the lower bound crosses certain size (from what I've seen it's about the 99% quantile of the past truncated Gaussian distribution) I get an Assertion error as well as other weird results such as distribution.var() = -204.544

#### Reproducing code example:

```
mu = 14.654
sigma = 2
b = np.Inf
a = 31.24
dist = truncnorm(a = (a-mu)/sigma, b = (b-mu)/sigma , loc = mu, scale = sigma)
print([dist.ppf(0.01), dist.ppf(0.99)])
```

#### Error message:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  ...
  File "/home/nitay/Phd/PETER/market_based_rl_agents/IPOMDP/TOM_zero_agents/tom_zero_manager_model.py", line 63, in select_best_response
    optional_bid, expected_payoff, _ = self.generate_optimal_bid()

  File "/home/nitay/Phd/PETER/market_based_rl_agents/IPOMDP/TOM_zero_agents/tom_zero_manager_model.py", line 51, in generate_optimal_bid
    grid = np.linspace(self.belief_distribution.ppf(0.01),

  File "/home/nitay/anaconda3/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py", line 455, in ppf
    return self.dist.ppf(q, *self.args, **self.kwds)

  File "/home/nitay/anaconda3/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py", line 2011, in ppf
    place(output, cond, self._ppf(*goodargs) * scale + loc)

  File "/home/nitay/anaconda3/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py", line 7163, in _ppf
    return _truncnorm_ppf(q, a, b)

  File "/home/nitay/anaconda3/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py", line 6933, in vf_wrapper
    return vf(*args)

  File "/home/nitay/anaconda3/lib/python3.7/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)

  File "/home/nitay/anaconda3/lib/python3.7/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)

  File "/home/nitay/anaconda3/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py", line 7092, in _truncnorm_ppf
    x = -_norm_ilogcdf((np.log1p(-q) + _norm_logsf(a)))

  File "/home/nitay/anaconda3/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py", line 7071, in _norm_ilogcdf
    z = z - (_norm_logcdf(z) - y) / _norm_logcdfprime(z)

  File "/home/nitay/anaconda3/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py", line 7050, in _norm_logcdfprime
    assert np.abs(z) > TRUNCNORM_TAIL_X/2

AssertionError
```

**Comment by nitayalon** [comment]
Scipy/Numpy/Python version information:
1.4.1 1.18.1 sys.version_info(major=3, minor=7, micro=6, releaselevel='final', serial=0)

**Comment by pvanmulbregt** [comment]
Hi @nitayalon,
This was fixed with the rewrite of `truncnorm` in gh-11393.  This should become available generally in the upcoming SciPy 1.5 release.
```
>>> mu = 14.654 
... sigma = 2 
... b = np.Inf 
... a = 31.24 
... dist = truncnorm(a = (a-mu)/sigma, b = (b-mu)/sigma , loc = mu, scale = sigma) 
... print([dist.ppf(0.01), dist.ppf(0.99)])                                                                                                                                                        
[31.242389836811416, 32.302030170356204]
```

**Comment by nitayalon** [comment]
Thanks a lot!
